server/container_remove.go

```diff
@@ -23,7 +23,7 @@ func (s *Server) RemoveContainer(ctx context.Context, req *pb.RemoveContainerReq
 	}
 
 	cState := s.runtime.ContainerStatus(c)
-	if cState.Status == oci.ContainerStateCreated || cState.Status == oci.ContainerStateRunning {
+	if cState.Status == oci.ContainerStateRunning {
```

Comment:
I think that we should be careful in making this change. If for e.g. one of the containers in a pod fails and kubelet determines that it needs to stop/remove the pod, it won't go and start all the containers in that pod that are still in created state.